### PR TITLE
meli: init at pre-alpha-0.3.2

### PR DESCRIPTION
**Pull request by 0x4A6F**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/applications/networking/mailreaders/meli/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  cargoSha256 = "0rlrffsbip3yz2z7zc39py4cdkw0dz0qdjprdr3l5g5wk9724qxb";
+
+  nativeBuildInputs = [ pkgconfig ];
+  buildInputs = [ openssl dbus ];
+
+  meta = with stdenv.lib; {
+    description = "Experimental terminal mail client aiming for configurability and extensibility with sane defaults";
+    homepage = "https://meli.delivery";
+    license = licenses.gpl3;
+    maintainers = [ maintainers."0x4A6F" ];
```
Comment (matthiasbeyer) [inline pkgs/applications/networking/mailreaders/meli/default.nix]: Feel free to add me here as well

**Review by matthiasbeyer — APPROVED** [review]

File: pkgs/applications/networking/mailreaders/meli/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+rustPlatform.buildRustPackage rec {
+  pname = "meli";
+  version = "pre-alpha-0.3.0";
+
+  src = fetchgit {
+    url = "https://git.meli.delivery/meli/meli.git";
+    rev = "${version}";
+    sha256 = "0slg9gz3800lxhfdqwlzqk3j9yi2yq7yl53dcyj6vhzk6kgz8lc2";
+  };
+  cargoPatches = [ ./cargo-lock.patch ];
```
Comment (alexarice) [inline pkgs/applications/networking/mailreaders/meli/default.nix]: An extra newline above this one would be nice for consistency

**Review by alexarice — APPROVED** [review]
Builds and runs with nix-review on nixos

File: pkgs/applications/networking/mailreaders/meli/default.nix (hunk 2 above)
Comment (0x4A6F) [reply pkgs/applications/networking/mailreaders/meli/default.nix]: Updated.

**Comment by alexarice** [comment]
Did you mean to request me to review this?

**Comment by alexarice** [comment]
@GrahamcOfBorg build meli

**Review by alexarice — COMMENTED** [review]
Doesn't seem to build on aarch so maybe `platforms` should be changed

**Comment by 0x4A6F** [comment]
@alexarice Should now build and run on aarch64.

**Comment by alexarice** [comment]
@GrahamcOfBorg build meli

**Review by alexarice — APPROVED** [review]

**Comment by aanderse** [comment]
@alexarice did you test the program locally?

**Comment by alexarice** [comment]
I think so but I'll do it again now

**Comment by alexarice** [comment]
Build fails with:
```
fatal: reference is not a tree: 7c2c8ae9453c81c026dde845aff5934a1df0de80
  Unable to checkout 7c2c8ae9453c81c026dde845aff5934a1df0de80 from https://git.meli.delivery/meli/meli.git.
```

**Comment by matthiasbeyer** [comment]
@0x4A6F please ff-pull from https://github.com/matthiasbeyer/nixpkgs/tree/master-meli - I fixed the hashes for the derivation.

Though, my build still fails, with:

```
[...]
building '/nix/store/h3chyx3vlf0icdxlpfhl1wa33wh91kg2-meli-pre-alpha-0.3.2.drv'...
unpacking sources
unpacking source archive /nix/store/gwcb2gsq1c5lfyfm7pdkcnldddgw77qn-meli-b7edec0
source root is meli-b7edec0
unpacking source archive /nix/store/fvxz7kh6z9jk40cq47rpap7dfnv0x1iy-meli-pre-alpha-0.3.2-vendor
patching sources
configuring
building
++ env CC_x86_64-unknown-linux-gnu=/nix/store/7hzmz83nv8khpwsk858yaac7y3idh0a7-gcc-wrapper-8.3.0/bin/cc CXX_x86_64-unknown-linux-gnu=/nix/store/7hzmz83nv8khpwsk858yaac7y3idh0a7-gcc-wrapper-8.3.0/bin/c++ CC_x86_64-unknown-linux-gnu=/nix/store/7hzmz83nv8khpwsk858yaac7y3idh0a7-gcc-wrapper-8.3.0/bin/cc CXX_x86_64-unknown-linux-gnu=/nix/store/7hzmz83nv8khpwsk858yaac7y3idh0a7-gcc-wrapper-8.3.0/bin/c++ cargo build --release --target x86_64-unknown-linux-gnu --frozen
error: the lock file /build/meli-b7edec0/Cargo.lock needs to be updated but --frozen was passed to prevent this
builder for '/nix/store/h3chyx3vlf0icdxlpfhl1wa33wh91kg2-meli-pre-alpha-0.3.2.drv' failed with exit code 101
error: build of '/nix/store/h3chyx3vlf0icdxlpfhl1wa33wh91kg2-meli-pre-alpha-0.3.2.drv' failed
```

**Comment by 0x4A6F** [comment]
The source branch *aarch64* vanished.
So I rebased the commit "meli: fix for aarch64" to [meli/51bb50abf3](https://git.meli.delivery/meli/meli/commit/51bb50abf38b4a6fc16acdb350fa3a889c6330da)

@matthiasbeyer Your commit is similar to "meli: init at pre-alpha-0.3.2", sans cargoPatches.

We might wait for the next release and thus get rid of cargoPatches and have aarch64 support?

**Comment by 0x4A6F** [comment]
Closing in favor of #76519.

**Pull request closed without merging**